File: linked-list/MyLinkedList.py
# python3 MyLinkedList.py

class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def addLast(self, value):
        new_node = Node(value)

        if self.head is None:
            self.head = new_node
        else:
            current = self.head

            while current.next:
                current = current.next

            current.next = new_node

    def addFirst(self, value):
        new_node = Node(value)

        if self.head is None:
            self.head = new_node
        else:
            new_node.next = self.head
            self.head = new_node

    def removeFirst(self):
        if self.head is not None:
            if self.head.next:
                self.head = self.head.next
            else:
                self.head = None

    def removeLast(self):
        if self.head is not None:

            if self.head.next:
                current = self.head

                while current.next.next:
                    current = current.next

                current.next = None
            else:
                self.head = None

    def getFirst(self):
        if self.head is not None:
            return self.head.value

        return None

    def getLast(self):
        if self.head is not None:
            current = self.head

            while current.next:
                current = current.next

            return current.value

        return None

    def display(self):
        if self.head is not None:
            current = self.head

            while current:
                print(f"[{current.value}]->")

                current = current.next

ll = LinkedList();

ll.addLast(4);
ll.addLast(3);
ll.addLast(2);
ll.addLast(1);
ll.addFirst(5);
ll.addFirst(6);
ll.addFirst(7);

ll.removeFirst();
ll.removeLast();

ll.display();

print(f"First Item: {ll.getFirst()}");
print(f"Last Item: {ll.getLast()}");

File: linked-list/MyLinkedList.swift
// swift MyLinkedList.swift

class Node<V> {
    var value: V
    var next: Node? = nil

    init(val: V) {
        value = val
    }
}

class LinkedList<T> {
    private var head: Node<T>? = nil

    func addLast(value: T) {
        let newNode = Node<T>(val:value)

        if head == nil {
            head = newNode
        } else {
            var current = head;

            while current?.next != nil {
                current = current?.next
            }

            current?.next = newNode;
        }
    }

    func addFirst(value: T) {
        let newNode = Node<T>(val: value)

        if head == nil {
            head = newNode
        } else {
            newNode.next = head;
            head = newNode;
        }
    }

    func removeFirst() {
        if head != nil {

            if head?.next == nil {
                head = nil
            } else {
                head = head?.next
            }
        }
    }

    func removeLast() {
        if head != nil {
            if head?.next == nil {
                head = nil
            } else {
                var current = head;

                while current?.next?.next != nil {
                    current = current?.next;
                }

                current?.next = nil;
            }
        }
    }

    func getFirst() -> T? {
        if let head {
            return head.value;
        }

        return nil;
    }

    func getLast() -> T? {
        if let head {
            var current = head;

            while current.next != nil {
                current = current.next!
            }

            return current.value
        }

        return nil;
    }

    func display() {
        var current = head;

        while current != nil {
            if let value = current?.value {
                print("[\(value)]-->")
            }

            current = current?.next
        }
    }
}


let ll = LinkedList<Int>()

ll.addLast(value: 4)
ll.addLast(value: 3)
ll.addLast(value: 2)
ll.addLast(value: 1)
ll.addFirst(value: 5)
ll.addFirst(value: 6)
ll.addFirst(value: 7)

ll.removeFirst()
ll.removeLast()

ll.display()

if let first = ll.getFirst() {
    print("First Item: \(first)")
}

if let last = ll.getLast() {
    print("Last Item: \(last)")
}

let ll2 = LinkedList<String>()

ll2.addLast(value: "cat");
ll2.addLast(value: "dog");
ll2.addFirst(value: "alligator");

ll2.display();

if let first = ll2.getFirst() {
    print("First Item: \(first)")
}

if let last = ll2.getLast() {
    print("Last Item: \(last)")
}